### PR TITLE
RavenDB-15530 upon TS conflict prefer deleted values

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Server/Documents/DocumentDatabase.cs [hunk 1]
```diff
@@ -458,11 +458,14 @@ private async Task ExecuteClusterTransactionTask()
                 var topology = ServerStore.LoadDatabaseTopology(Name);
                 if (topology.Promotables.Contains(ServerStore.NodeTag))
                 {
-                    await Task.Delay(1000);
+                    await Task.Delay(1000, DatabaseShutdown);
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/DocumentDatabase.cs:461]: That can throw `TaskCancelledException`. Is it ok? I see the code below just catches everything.

File: src/Raven.Server/Documents/DocumentDatabase.cs [hunk 2]
```diff
@@ -458,11 +458,14 @@ private async Task ExecuteClusterTransactionTask()
...
+                    if (DatabaseShutdown.IsCancellationRequested)
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/DocumentDatabase.cs]: Whats the point of checking it here since we're passing `DatabaseShutdown` one line above?

Comment (karmeli87) [reply src/Raven.Server/Documents/DocumentDatabase.cs]: remvoed

File: src/Raven.Server/Documents/DocumentDatabase.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/DocumentDatabase.cs:461]: `TaskCancelledException` is caught higher in the stack by `catch (OperationCanceledException)`

**Review by arekpalinski — APPROVED** [review]

**Review by aviv86 — APPROVED** [review]

**Pull request merged**